pkgs/development/python-modules/pypdfium2/default.nix

```diff
@@ -0,0 +1,139 @@
+{
+  lib,
+  pkgs,
+  buildPythonPackage,
+  fetchFromGitHub,
+  fetchurl,
+
+  # build-system
+  setuptools-scm,
+
+  # dependencies
+  setuptools,
+  pdfium-binaries,
+  numpy,
+  pillow,
+
+  # tests
+  pytest,
+  python,
+}:
+
+let
+  pdfiumVersion = "${pdfium-binaries.version}";
+
+  headers = fetchurl {
+    url = "https://pdfium.googlesource.com/pdfium/+archive/refs/heads/chromium/${pdfiumVersion}/public.tar.gz";
+    hash = "sha256-yeuIk2Dp36cVVJX4/3OTq769wSJB0QhuZfU5lDyDotQ=";
+  };
+
+  # They demand their own fork of ctypesgen
+  ctypesgen = buildPythonPackage rec {
+    pname = "ctypesgen";
+    version = "1.1.1+g${src.rev}"; # the most recent tag + git version
+
+    src = fetchFromGitHub rec {
+      inherit pname;
+      version = rev;
+      owner = "pypdfium2-team";
+      repo = "ctypesgen";
+      rev = "848e9fbb1374f7f58a7ebf5e5da5c33292480b30";
+      hash = "sha256-3JA7cW/xaEj/DxMHEypROwrKGo7EwUEcipRqALTvydw=";
+    };
+
+    pyproject = true;
```

Comment:
Please make sure these attributes are sorted in the same way as the main derivation, for consistency.